include/aca_zeta_programming.h

```diff
@@ -30,20 +32,43 @@ struct zeta_config {
   uint32_t port_inband_operation;
 };
 
+struct aux_gateway_entry {
+  uint oam_port;
+  uint group_id;
+};
+
 class ACA_Zeta_Programming {
   public:
   static ACA_Zeta_Programming &get_instance();
 
+  uint get_or_create_group_id(string auxGateway_id);
+
   int create_or_update_zeta_config(const alcor::schema::AuxGateway current_AuxGateway,
                                    const string vpc_id, uint32_t tunnel_id);
 
   int delete_zeta_config(const alcor::schema::AuxGateway current_AuxGateway,
                          const string vpc_id, uint32_t tunnel_id);
 
+  bool is_exist_group_rule(uint group_id);
+
+  uint get_oam_server_port(string auxGateway_id);
+
+  void set_oam_server_port(string auxGateway_id, uint port_number);
+
+  bool is_exist_oam_port(uint port_number);
+
   private:
   int _create_or_update_zeta_group_entry(zeta_config *zeta_config_in);
 
   int _delete_zeta_group_entry(zeta_config *zeta_config_in);
+
+  private:
+  // unordered_map<aux_gateway_id, aux_gateway_entry>
+  unordered_map<string, aux_gateway_entry> _zeta_gateways_table;
```

Comment:
do we need another hashtable? can we keep this information in the existing _vpcs_table since zeta gateway info is assoicated with VPC.